lib/classes/Utils.js

```diff
@@ -142,152 +143,167 @@ class Utils {
+          };
+
+          memorySizeAndTimeoutPerFunction.push(memorySizeAndTimeoutObject);
+        });
+      }
 
-    const auth = `${writeKey}:`;
+      // event related information retrieval
+      const numberOfEventsPerType = [];
+      const eventNamesPerFunction = [];
+      if (numberOfFunctions) {
+        _.forEach(serverless.service.functions, (func) => {
+          if (func.events) {
+            const funcEventsArray = [];
+
+            func.events.forEach((event) => {
+              const name = Object.keys(event)[0];
+              funcEventsArray.push(name);
+
+              const alreadyPresentEvent = _.find(numberOfEventsPerType, { name });
+              if (alreadyPresentEvent) {
+                alreadyPresentEvent.count++;
+              } else {
+                numberOfEventsPerType.push({
+                  name,
+                  count: 1,
+                });
+              }
+            });
+
+            eventNamesPerFunction.push(funcEventsArray);
+          }
+        });
+      }
 
-    const data = {
-      userId,
-      event: 'Serverless framework usage',
-      properties: {
-        command: {
-          name: serverless.processedInput.commands.join(' '),
-          isRunInService: (!!serverless.config.servicePath),
-        },
-        service: {
-          numberOfCustomPlugins: _.size(serverless.service.plugins),
-          hasCustomResourcesDefined,
-          hasVariablesInCustomSectionDefined: (!!serverless.service.custom),
-          hasCustomVariableSyntaxDefined,
-        },
-        provider: {
-          name: serverless.service.provider.name,
-          runtime: serverless.service.provider.runtime,
-          stage: serverless.service.provider.stage,
-          region: serverless.service.provider.region,
-        },
-        functions: {
-          numberOfFunctions,
-          memorySizeAndTimeoutPerFunction,
-        },
-        events: {
-          numberOfEvents: numberOfEventsPerType.length,
-          numberOfEventsPerType,
-          eventNamesPerFunction,
-        },
-        general: {
-          userId,
-          timestamp: (new Date()).getTime(),
-          timezone: (new Date()).toString().match(/([A-Z]+[\+-][0-9]+.*)/)[1],
-          operatingSystem: process.platform,
-          serverlessVersion: serverless.version,
-          nodeJsVersion: process.version,
+      let hasCustomResourcesDefined = false;
+      // check if configuration in resources.Resources is defined
+      if ((serverless.service.resources &&
+        serverless.service.resources.Resources &&
+        Object.keys(serverless.service.resources.Resources).length)) {
+        hasCustomResourcesDefined = true;
+      }
+      // check if configuration in resources.Outputs is defined
+      if ((serverless.service.resources &&
+        serverless.service.resources.Outputs &&
+        Object.keys(serverless.service.resources.Outputs).length)) {
+        hasCustomResourcesDefined = true;
+      }
```

Comment:
I think this if statement and the one before it can be grouped into one. Maybe put all those `&&` into meaningful variable names for refactoring.